identity-did/src/service/bitmap_revocation_service.rs

```diff
@@ -0,0 +1,144 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use identity_core::common::Object;
+
+use super::BitmapRevocationEndpoint;
+use super::Service;
+use super::ServiceEndpoint;
+use crate::did::DIDUrl;
+use crate::did::DID;
+use crate::error::Error;
+use crate::error::Result;
+
+pub const BITMAP_REVOCATION_METHOD: &str = "RevocationBitmap2022";
+
+/// A DID Document Service used to enable validators to check the status of a credential.
+#[derive(Clone, Debug, PartialEq)]
+pub struct BitmapRevocationService<D: DID + Sized> {
+  pub(crate) id: DIDUrl<D>,
+  pub(crate) type_: String,
+  pub(crate) service_endpoint: BitmapRevocationEndpoint,
+}
+
+impl<D: DID + Sized> BitmapRevocationService<D> {
+  /// Creates a new [`BitmapRevocationService`].
+  pub fn new(id: DIDUrl<D>, service_endpoint: BitmapRevocationEndpoint) -> Result<Self> {
+    if id.fragment().unwrap_or_default().is_empty() {
+      return Err(Error::InvalidService("invalid id - missing fragment"));
+    }
```

Comment:
I'm not sure how useful a `BitmapRevocationService` is anymore. It's not used to enforce any (de)serialization invariants and `BitmapRevocationEndpoint` could be replaced by `RevocationBitmap` directly in most places.